### PR TITLE
Format with black

### PR DESCRIPTION
**Pull request by willbarton**
This commit has had [`black`](https://github.com/psf/black), the autoformatter, run on the `flags` package and `setup.py`

It also adds `black --check` to the `tox` `lint` environment and adds a section in `CONTRIBUTING.md` about code style and tools.

### CODE REVIEW

File: flags/conditions.py [hunk 1]
```diff
@@ -64,23 +65,25 @@ def boolean_condition(condition, **kwargs):
...
 
-@register('user')
+@register("user")
 def user_condition(username, request=None, **kwargs):
     """ Does request.user match the expected username? """
     if request is None:
-        raise RequiredForCondition("request is required for condition "
-                                   "'user'")
+        raise RequiredForCondition(
+            "request is required for condition " "'user'"
```
Comment (higs4281) [inline flags/conditions.py]:
```suggestion
            "request is required for condition 'user'"
```

File: flags/conditions.py [hunk 2]
```diff
@@ -64,23 +65,25 @@ def boolean_condition(condition, **kwargs):
...
-@register('anonymous')
+@register("anonymous")
 def anonymous_condition(boolean_value, request=None, **kwargs):
     """ request.user an anonymous user, true or false based on boolean_value
     """
     if request is None:
-        raise RequiredForCondition("request is required for condition "
-                                   "'anonymous'")
+        raise RequiredForCondition(
+            "request is required for condition " "'anonymous'"
```
Comment (higs4281) [inline flags/conditions.py]:
```suggestion
            "request is required for condition 'anonymous'"
```

File: flags/conditions.py [hunk 3]
```diff
@@ -93,31 +96,33 @@ def anonymous_condition(boolean_value, request=None, **kwargs):
...
 
-@register('parameter')
+@register("parameter")
 def parameter_condition(param_name, request=None, **kwargs):
     """ Is the parameter name part of the GET parameters? """
     if request is None:
-        raise RequiredForCondition("request is required for condition "
-                                   "'parameter'")
+        raise RequiredForCondition(
+            "request is required for condition " "'parameter'"
```
Comment (higs4281) [inline flags/conditions.py]:
```suggestion
            "request is required for condition 'parameter'"
```

File: flags/conditions.py [hunk 4]
```diff
@@ -93,31 +96,33 @@ def anonymous_condition(boolean_value, request=None, **kwargs):
...
 
-@register('path matches')
+@register("path matches")
 def path_condition(pattern, request=None, **kwargs):
     """ Does the request's path match the given regular expression? """
     if request is None:
-        raise RequiredForCondition("request is required for condition "
-                                   "'path'")
+        raise RequiredForCondition(
+            "request is required for condition " "'path'"
```
Comment (higs4281) [inline flags/conditions.py]:
```suggestion
            "request is required for condition 'path'"
```

File: tox.ini [hunk 5]
```diff
@@ -25,10 +24,12 @@ deps=
 [testenv:lint]
 basepython=python3.6
 deps=
+    black
     flake8>=2.2.0
     isort>=4.2.15
 commands=
-    flake8 .
+    black --check flags setup.py
+    flake8 flags setup.py
```
Comment (Scotchester) [inline tox.ini]: Why specify `flags setup.py`? Just to avoid looking over a bunch of non-Python files?

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+[tool.black]
+line-length = 78
```
Comment (Scotchester) [inline pyproject.toml]: If we're gonna take this plunge, maybe it's time to reconsider our line length limit? [Their logic in selecting a default for Black of 88](https://black.readthedocs.io/en/stable/the_black_code_style.html#line-length) seems sound.

**Comment by willbarton** [comment]
> I have no big objection, but the vast majority of black-enforced changes are single-quote/double-quote swaps, which I see little value in

Yeah, black is highly opinionated about quotes. There's a switch to ignore single/double quotes, but I feel like if we're going to embrace an auto formatter, we should embrace it.

File: pyproject.toml (hunk 6 above)
Comment (willbarton) [reply pyproject.toml]: Sure. That'd just mean we have to configure `flake8` to allow it.

My thought it configuring `black` to limit the line length is that it requires less change-your-code-editor churn for us.

**Comment by willbarton** [comment]
> Re: quote normalization, we coooould skip it 😁

👇 

> I feel like if we're going to embrace an auto formatter, we should embrace it.

File: pyproject.toml (hunk 6 above)
Comment (Scotchester) [reply pyproject.toml]: Exactly :) Would be nice to get a little line length bonus out of this transition. But even if we don't, this should be set to 79, no?

Comment (willbarton) [reply pyproject.toml]: Then I'd have to increase the default size of my vim splits. My cheese is moving too much!

But yes, it should be 79. I had been seeing some flake8 errors that I had thought were related to 79, but they aren't.

File: tox.ini (hunk 5 above)
Comment (willbarton) [reply tox.ini]: Basically.

**Pull request merged**